app/lib/pdf/finance/invoice_generator.rb

```diff
@@ -76,13 +71,17 @@ def print_header
         move_down(3)
       end
 
+      def print_logo
+        @data.with_logo do |logo|
+          @pdf.image(logo, fit: [200,50], position: :left) if logo
+        end
+      end
+
       def print_address_columns
         # TODO: cleanup the constants
         two_columns( [0.mm, @pdf.cursor], height: 50.mm) do |column|
-          case column
-          when :left then print_address( @data.provider, 'Issued by')
-          when :right then print_address( @data.buyer, 'For')
-          end
+          print_address( @data.provider, 'Issued by') if column == :left
+          print_address( @data.buyer, 'For') if column == :right
```

Comment:
We have some extra spaces in here for the `two_columns` and `print_address` invocations.